### PR TITLE
First pull request

### PR DESCRIPTION
**Pull request by netliteIT**
Python 3.8 tested
Log in syslog
Log in utc
socket try except 
decode iso8559-1
Zebra Errors and Warnings

### CODE REVIEW

File: zpl/printer.py [hunk 1]
```diff
@@ -81,56 +199,117 @@ def get_printer_config(self, reload=False):
...
+            print('[' + str(time.process_time()) + ']' + text)
+
+    def __init__(self, args, host, port=9100, socket_timeout=5):
+        try:
+            start = time.time()
+            self.log(args, syslog.LOG_INFO, 'Socket create:' + host + ':' + str(port))
+            self.socket = socket.create_connection((host, port), timeout=socket_timeout)
+        except socket.timeout:
+            self.log(args, syslog.LOG_EMERG, 'Error:Socket Timeout')
+            exit(1)
```
Comment (cod3monk) [inline zpl/printer.py]: Better raise here as well, this should not exit whichever program is using it.

File: zpl/printer.py [hunk 2]
```diff
@@ -81,56 +199,117 @@ def get_printer_config(self, reload=False):
...
+
     def send_job(self, zpl2):
-        self.socket.sendall(zpl2)
-    
+        try:
+            syslog.syslog(syslog.LOG_INFO,'Send:' + zpl2)
+            self.socket.sendall(zpl2.encode('iso8859-1'))
+        except socket.timeout:
+            syslog.syslog(syslog.LOG_CRIT,'Error:Socket Timeout')
+            exit(1)
```
Comment (cod3monk) [inline zpl/printer.py]: no exit, just raise.

**Review by cod3monk — CHANGES_REQUESTED** [review]

**Comment by netliteIT** [comment]
There's also a sleep of 0.2 because while I was writing a lowlevel socket communication with Zebra printers I noticed that sometimes answers to huge request (eg. the complete configuration xml) comes with a little delay. 
I've configured timeouts in socket layer because it happens that communication doesn't ends correctly with \x03.
Don't know if you agree avout timeout and delay.

**Comment by cod3monk** [comment]
The timeout makes sense to me. Looks good :) I'll make a version number increase and push it out to pypi as well.

Thanks for the PR!

**Pull request merged**